pkg/dataplane/reloadCredentials.go

```diff
@@ -0,0 +1,154 @@
+package dataplane
+
+import (
+	"context"
+	"errors"
+	"log"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/msi-dataplane/pkg/dataplane/swagger"
+	"github.com/fsnotify/fsnotify"
+)
+
+var (
+	// Errors returned when reloading credentials
+	errCreateFileWatcher = errors.New("failed to create file watcher")
+	errAddFileToWatcher  = errors.New("failed to add credentialFile to file watcher")
+	errLoadCredentials   = errors.New("failed to load credentials from file")
+)
+
+type reloadingCredential struct {
+	currentValue *azidentity.ClientCertificateCredential
+	cloud        string
+	lock         *sync.RWMutex
+	logger       *log.Logger
```

Comment:
Oops, sorry sorry sorry - I was not specific enough. The `logr.Logger` interface is correct here. I just meant that our default implementation should use the standard library so we don't bring in unnecessary dependencies. Structured logging is the way to go. See: https://github.com/go-logr/logr?tab=readme-ov-file#slog-interoperability